### PR TITLE
Add Boolean assertion transform functions.

### PR DESCRIPTION
**Pull request by shenyu0127**
This PR adds the following transform functions:
- IS TRUE
- IS NOT TRUE
- IS FALSE
- IS NOT FALSE

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsFalseTransformFunction.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.common.function.TransformFunctionType;
+import org.apache.pinot.core.operator.ColumnContext;
+
+
+public class IsFalseTransformFunction extends IsTrueTransformFunction {
```
Comment (gortiz) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsFalseTransformFunction.java:27]: I don't think it is a good idea to extend `IsTrueTransformFunction` in order to completely change the semantic.

Instead we should have some kind of `AbstractIsBooleanTransformFunction` that `IsTrueTransformFunction`, `IsFalseTransformFunction`, `IsNotFalseTransformFunction`, etc extend

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsTrueTransformFunction.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  @Override
+  public String getName() {
+    return TransformFunctionType.IS_TRUE.getName();
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, ColumnContext> columnContextMap) {
+    Preconditions.checkArgument(arguments.size() == 1, "Exact 1 argument is required for IS_NULL");
+    _transformFunction = arguments.get(0);
```
Comment (gortiz) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsTrueTransformFunction.java:43]: This is more a question to understand our code than something to change in this PR. I guess all these functions require to have an argument whose type is boolean. Where do we enforce that type constraint?

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsFalseTransformFunction.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsFalseTransformFunction.java:27]: +100 to the abstract class. Four implementations should just implement the Boolean operations.

Comment (shenyu0127) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsFalseTransformFunction.java]: Done.

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsTrueTransformFunction.java (hunk 2 above)
Comment (shenyu0127) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsTrueTransformFunction.java]: In V1 transform functions we use integer to represent boolean and use `BaseTransformFunction` to do casting. So we don't need to enforce the type constraint.

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseBooleanAssertionTransformFunction.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    if (_nullHandlingEnabled) {
+      nullBitmap = _transformFunction.getNullBitmap(valueBlock);
+    }
+    if (nullBitmap != null) {
+      for (int docId = 0; docId < length; docId++) {
+        if (nullBitmap.contains(docId)) {
+          if (returnsTrueWhenValueIsNull()) {
+            _intValuesSV[docId] = 1;
+          }
+        } else if (intValuesSV[docId] != 0) {
```
Comment (xiangfu0) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseBooleanAssertionTransformFunction.java:61]: logic of line 61 to 67 and 71 to 77 are same and we can also combine `returnsTrueWhenValueIsTrue` and `returnsTrueWhenValueIsFalse` into  one method like
```
if (evalBooleanOp()) {
  _intValuesSV[docId] = 1;
}
```

Comment (shenyu0127) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseBooleanAssertionTransformFunction.java]: Done.

Good suggestion!

**Review by xiangfu0 — APPROVED** [review]

**Pull request merged**